### ISSUE TITLE
[Code health] Eliminate forked BottomSheetBehavior class and simplify impl

### ISSUE THREAD

**Issue by gino-m** [google/ground-android#459]
We made a copy of the BottomSheetBehavior class to allow us to specify set the `expandedOffset` such that the header gets obscured behind the toolbar. Material 1.1.0 allows us to set the offset, but only to a non-negative number. To avoid needing to copy and modify the implementation again, looking for a simpler solution.

I've tried a dozen or so variants, and the _least_ hacky and most succinct is shared here:
https://github.com/gino-m/slidey-demo/blob/master/app/src/main/java/com/google/android/slidey/MainActivity.java
https://github.com/gino-m/slidey-demo/blob/master/app/src/main/java/com/google/android/slidey/FloatingHeaderBehavior.java

![slidey](https://user-images.githubusercontent.com/228050/80333450-1621d800-881c-11ea-9f4e-7c73cbe8afb9.gif)

Blocking #458.

@dturner fyi. @cketcham would love to hear what you think of this sol'n when/if you have a sec!

**Comment by cketcham** [comment]
Seems like a good solution which uses coordinator behaviors to keep things contained.

**Comment by dturner** [comment]
@gino-m Thanks very much for putting the demo together. Really clear what's going on now. It looks like you fixed the touch propagation issue from header->bottomsheet in `onLayoutChild` which is great. This seems like a good solution given the absence of anything simpler. 

FYI I spoke to Chris Banes to check whether there was any simpler way and he said there isn't and your `FloatingHeaderBehaviour` should be fine.

**Comment by gino-m** [comment]
@cketcham @dturner thanks for the feedback. I'll work on integrating this into the Ground bottom sheet in a new PR soon.

**Comment by gino-m** [comment]
@cketcham @dturner I'm coming up against another hard problem - how to correctly handle touch events on the header, and the transfer them to the BottomSheetBehavior. I've tried creating a TouchDelegate, but that doesn't seem to have any effect when set on the View when using BottomSheetBehavior. 

The problem with the example linked above is that all touch events are dispatched to the bottom sheet view, including click events, which can either be taps, or startDrag as interpreted by the BottomSheetBehavior's ViewDragHelper. We could use a ViewDragHelper in the header and scroll the bottom sheet directly, but then we'd lose all the inertia, fling, and snap functionality of bottom sheet. Also, dispatching the motion events of one view without updating the x,y is wrong, since the x,y is relative to the origin of a different view, and may not even fit within the bounds of the bottom sheet.

I've also tried placing a FrameLayout inside the bottom sheet view, positioning it over the header with clipping off and negative translationY, but the touch events are ignored :/.

Wondering now if allowing a negative `expandedOffset` in `BottomSheetBehavior` wasn't actually a better solution after all. Any ideas on how we can get the header to act as a valid drag target for the bottom sheet, without it needing to be in the bottom sheet view?
